tools/ci_build/build.py

```diff
@@ -2107,10 +2097,21 @@ def main():
     rocm_home = setup_rocm_build(args, configs)
 
     if args.update or args.build:
-        os.makedirs(build_dir, exist_ok=True)
+        for config in configs:
+            os.makedirs(get_config_build_dir(build_dir, config), exist_ok=True)
 
     log.info("Build started")
+
     if args.update:
+        if is_reduced_ops_build(args):
+            from reduce_op_kernels import reduce_ops
+            for config in configs:
+                reduce_ops(
+                    config_path=args.include_ops_by_config,
+                    build_dir=get_config_build_dir(build_dir, config),
```

Comment:
If someone want to run the reduction script before building, then the script will need the exact location of the reduced files? See this comment of my previous PR, [https://github.com/microsoft/onnxruntime/pull/9908#discussion_r761767217](https://github.com/microsoft/onnxruntime/pull/9908#discussion_r761767217)

Personally I think it's ok we can disable the reduction beforehead completely, or we can add anb extra param to the build script to allow a custom folder for reduced cc/inc files?

Reply:
The idea was to always put the reduced files under the build directory. Users will have to specify the build directory anyway. Is there a good reason for allowing other locations?

Running the op reduction separately may be useful when building without build.py. In that case, the same build directory should be passed to reduce_op_kernels.py and CMake.

Updating reduce_op_kernels.py to automatically create the build directory if it doesn't exist might be nicer, I can do that.